### PR TITLE
Cleanup type hints + spelling, source API links

### PR DESCRIPTION
**Pull request by ryancoe**
## Description
As I am using the fancy new VS Code IDE, I was made aware of a couple minor mistakes in our type hints and spelling typos.

- [x] Type Hints errors
- [x] Type Hints using `Optional` type (https://docs.python.org/3/library/typing.html#typing.Optional)
- [x] Spelling
- [x] Source API links in docs (https://www.sphinx-doc.org/en/master/usage/restructuredtext/domains.html#cross-referencing-python-objects)

## Checklist for PR
- [x] Authors read the [contribution guidelines](https://github.com/SNL-WaterPower/WecOptTool/blob/main/.github/CONTRIBUTING.md)
- [x] The pull request is from an issue branch (not main) on your fork to the [main branch in WecOptTool](https://github.com/SNL-WaterPower/WecOptTool). 
- [x] The authors have given the admins edit access
- [x] All changes adhere to the style guide including PEP8, Docstrings, and Type Hints.
~~Added note in [Changelog](https://github.com/SNL-WaterPower/WecOptTool/blob/main/CHANGES.md)~~
~~Modified the documentation if applicable~~
~~Modified or added a new test~~
- [x] All tests pass
- [x] Documentation builds
~~[Reference or close any relevant issues](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue)~~

### CODE REVIEW

File: wecopttool/core.py [hunk 1]
```diff
@@ -1226,7 +1226,7 @@ def delaxes(axs, idof, jdof, ndof):
 
 
 def post_process_continuous_time(results: xr.DataArray
-                                 ) -> Callable[float, float]:
+                                 ) -> Callable[[float, float]]:
```
Comment (cmichelenstrofer) [inline wecopttool/core.py:1229]: The resulting function has input `float` and output `float`. I think the original line was correct. I think ` Callable[[float, float]]` means two float inputs and no outputs. Am I wrong?

File: examples/tutorial_1_wavebot.ipynb [hunk 2]
```diff
@@ -33,7 +33,7 @@
    "source": [
     "# logging\n",
     "import logging\n",
-    "logging.basicConfig(level=logging.INFO)\n",
+    "logging.basicConfig(level=logging.ERROR)\n",
```
Comment (cmichelenstrofer) [inline examples/tutorial_1_wavebot.ipynb:36]: I think for the first example at least, the INFO level was good. No?

**Review by cmichelenstrofer — APPROVED** [review]
That was a lot of typos! Maybe I don't have spell check on in VS Code.

File: examples/tutorial_1_wavebot.ipynb (hunk 2 above)
Comment (ryancoe) [reply examples/tutorial_1_wavebot.ipynb]: yeah, that makes sense. With c9b1eacdde8e61f2bc398ecd53a36cdab019611a I've returned it to `INFO` and added a hyperlink to the website explaining logging levels (https://docs.python.org/3/library/logging.html#levels).

File: wecopttool/core.py (hunk 1 above)
Comment (ryancoe) [reply wecopttool/core.py]: Yes, you're right. The correct syntax in this case is `Callable[[float], float]`. Updated w. 6c922475824c76cd0eadec5b83fb64241b390032.

https://docs.python.org/3.6/library/typing.html#typing.Callable

**Comment by ryancoe** [comment]
I'm going to hold off until #53 is merged and then update `type | None` to `Optional[type]`

**Pull request merged**